crypto/x509/by_file.c

```diff
@@ -123,6 +121,8 @@ int X509_load_cert_file(X509_LOOKUP *ctx, const char *file, int type)
         X509err(X509_F_X509_LOAD_CERT_FILE, X509_R_BAD_X509_FILETYPE);
         goto err;
     }
+    if (ret == 0)
+        X509err(X509_F_X509_LOAD_CERT_FILE, X509_R_EMPTY_FILE);
```

Comment:
That's not quite true.  This happens when the file doesn't contain a cert, but that doesn't mean the file is empty.  `X509_R_NO_CERTIFICATE_FOUND` would be better.